### ISSUE TITLE
Messages take 1-15 minutes to deliver, sometimes longer for photos. 

### ISSUE THREAD

**Issue by miserablesufferer** [signalapp/Signal-iOS#5587]
This issue happens on iOS, macOS, and android. I'm using the latest version of the app, by the way. 
this happens with everyone I text. 
I'm not sure what other information too add, I'm using mac os high sierra and iOS 15.6.1, although I don't think that has anything to do with the issue.
I really love the app otherwise, but this makes it practically useless for casual conversation, and I'm gonna have to use something else, not even sure what. 
It also happens with texts people send me sometimes. 
Please fix this
I'm happy to answer any questions.

**Comment by bengiv** [comment]
Big still existing in newest version, please fix

**Comment by Miguel-Signal** [comment]
Hi @miserablesufferer @bengiv 

If you see ![image](https://github.com/signalapp/Signal-iOS/assets/96307097/cce3d752-5f70-4e57-8f7d-4802234115d7) (Sent) after you send a message this indicates that your message has been sent to the Signal service and there is no issue with your phone's connectivity.

The issue is mostly likely with notifications on the recipients end. If your recipients are using an iPhone, could you ask them to re-register for push notifications to see if this helps resolves the issue?

Signal Settings  > Notifications > Re-register Push Notifications > OK.

Please let me know if this is successful.

**Comment by bengiv** [comment]
Hi @Miguel-Signal 
In my case the recipient unfortunately uses an Android Device, is there any other way to try to resolve the issue?

**Comment by Miguel-Signal** [comment]
Could you please instruct the recipient to email us at support@signal.org and reference this Github issue so we can troubleshoot with them directly?

**Comment by MuteBefehl** [comment]
@Miguel-Signal You should have received an email from me regarding this issue with @bengiv.

**Comment by maurice1** [comment]
Experiencing this too with a few contacts using Signal on iOS (I am on Android). Even now after 12 days the message is still not delivered.

Isn't there a possibility to re-register push notifications automatically, for example after app updates?

**Comment by bme13** [comment]
Noticed this when having a chat from an Android device to an IOS device. There were multiple messages (01/28) around 13:39 (CET) and at 13:41 the message I send (from Android device) just got one check. 
Trying to find this in the attached IOS logfile it seems a few type indicators are received at 12:41 but then the data is received at 14:20.
https://debuglogs.org/ios/7.43.0/38a174ff7e4198d1bb05548e5131daeebea613db461f26dd4f791383add83293.zip.

**Comment by Delamcode** [comment]
I am currently encountering this with my own receiving of notifications. LPM is off, background refresh on, all network is connected. Sometimes I’ll close the app (open in background) and only get the notification 2min after *my app* says it was sent by my contact.

**Comment by bme13** [comment]
While in a conversation all of a sudden it took nearly 3 minutes to deliver a message.
1st message 1747833619109 (13:20:19:109) received 13:20:19:292  ok
2nd message 1747833657003 (13:20:57:003) received 13:23:50:521 delayed (single checkmark)
https://debuglogs.org/ios/7.58.0/0f947c721f11ebfe6e33122e5d621debd176be1dd2af59fecc5d37d07ccf1514.zip

**Comment by bme13** [comment]
Today this happened again. While in an conversation it took ~3min to deliver a message. Interestingly this was in a small group where besides me there are 2 other people, one with an iPhone and one with an Android. 
1st message from me 1748083658436 (12:47:38) ok
12:47:39 delivery receipt receiptid 15
12:47:41 delivery receipt receiptid 6
Then I receive a message from one of the participants 
2nd message from me 1748083768329 (12:49:28)
12:52:38 delivery receipts receiptid 15
12:52:41 delivery receipts receiptid 6

So both the Android and IOS were delayed leading me to think that this might be server related in this case. I can't report it in the server repo so just add it here. Also dropping my Android log since it might contain additional information.
https://debuglogs.org/android/7.43.1/d5d1f4ac0d95a271d2421bc6e373c2143a4f67b49b7be156e1608d2478cd1185

**Comment by bme13** [comment]
Start a conversation from Android to IOS user. The first message went through fine, got a response. Send a second message from Android to IOS and that was not delivered. The IOS user was downstairs so after 3min I went down and asked to open Signal. That resulted in receiving this message. This kind of behaviour seems to happen with both IOS and Android. Not sure if this is a client issue or server related. See https://github.com/signalapp/Signal-Android/issues/13975 for the related Android issue.
First message: 1749641198732 this one was ok
Second message: 1749641250770 this one only delivered after opening
https://debuglogs.org/ios/7.60.0/ae471d5cdf206203a6b74af728acfb8441384fc61718871084f5421ec1ab2abf.zip

**Comment by bme13** [comment]
Another exact occurrence. Start a conversation. First message ok, get a response. Send a response to that and that one keeps having one check for more than 20 min. Then as a test put a reaction to a message and the message gets delivered.

![Image](https://github.com/user-attachments/assets/e645f890-0f03-4f22-b623-10f688a1b262)
Message that got delayed: 1750491006613
Message I've reacted on and that got the ball rolling again: 1750490975927
https://debuglogs.org/ios/7.63.0/4859a8a517c1779df71b8c533f7113c3cc8fcb50888c52d99aafdd10d39958c5.zip
btw. Is someone actually seeing these logs and are they helpful? It starts to become a bit annoying to keep requesting logfiles.

**Comment by bme13** [comment]
Another occurrence. 
delayed message: 1754582517925 (and a few consecutive messages)
message that got things started again: 1754583018333
https://debuglogs.org/ios/7.70.1/3a3c401de66aef94a45eaa6fe50785d2de4dbfdc4c3f79669f273d12bcaa9388.zip

**Comment by Miguel-Signal** [comment]
Thanks @bme13, could you confirm what Signal version the recipient was on when this latest occurrence happened?

**Comment by bme13** [comment]
It's version 7.70.1 (917)

**Comment by Miguel-Signal** [comment]
Thanks for confirming. Could you share another log link from the recipient?

**Comment by bme13** [comment]
Yes, here's one just created: https://debuglogs.org/ios/7.70.1/2cbed19335cafcf7bc2db6eb1566bd5fb7427a243177c6946538c92c13b7591d.zip

**Comment by Miguel-Signal** [comment]
Thanks again. I have shared these logs for review. I may reach back out to you with more questions.

We appreciate your help in reporting this and improving Signal for everyone!

**Comment by max-signal** [comment]
We made some changes in August 2025 (culminating in bd3363ce939d7e1e20b5e3b6de7cbdbee19a95e0) that were meant to address message delivery reliability problems. Given that this thread has been quiet since then, I’m going to close it as resolved, but please do reach out if you experience delivery problems on the latest version so that we can investigate.
